src/use/crypto/signing.ts

```diff
@@ -0,0 +1,140 @@
+import { arrayBufferToBase64, base64ToArrayBuffer, signingAlgorithm } from "../crypto/certificates";
+import { ec } from "elliptic";
+import BN from "bn.js";
+const Signature = require("elliptic/lib/elliptic/ec/signature.js");
+import * as asn1js from "asn1js";
+
+// map for easy lookup of the "N/2" and "N" value per elliptic curve
+const p256ec = new ec("p256");
+const p384ec = new ec("p384");
+
+const ordersForCurve = {
+    secp256r1: {
+        halfOrder: p256ec.n?.shrn(1),
+        order: p256ec.n,
+    },
+    secp384r1: {
+        halfOrder: p384ec.n?.shrn(1),
+        order: p384ec.n,
+    },
+};
+
+export function preventMalleability(signature: { r: Uint8Array; s: Uint8Array }, curveParams: { name: string }) {
+    let halfOrder = undefined;
+    switch (curveParams.name) {
+        case "P-256":
+            halfOrder = ordersForCurve.secp256r1.halfOrder;
+            break;
+        case "P-384":
+            halfOrder = ordersForCurve.secp384r1.halfOrder;
+            break;
+        default:
+            break;
+    }
+
+    let order = undefined;
+    switch (curveParams.name) {
+        case "P-256":
+            order = ordersForCurve.secp256r1.order;
+            break;
+        case "P-384":
+            order = ordersForCurve.secp384r1.order;
+            break;
+        default:
+            break;
+    }
+
+    if (!(halfOrder && order)) {
+        throw new Error(
+            'Can not find the half order needed to calculate "s" value for immalleable signatures. Unsupported curve name: ' +
+                curveParams.name
+        );
+    }
+
+    const bnS = new BN(signature.s);
+
+    // in order to guarantee 's' falls in the lower range of the order, as explained in the above link,
+    // first see if 's' is larger than half of the order, if so, it needs to be specially treated
+    if (bnS.cmp(halfOrder) === 1) {
+        // module 'bn.js', file lib/bn.js, method cmp()
+        // convert from BigInteger used by jsrsasign Key objects and bn.js used by elliptic Signature objects
+        const bigNum = order;
+        const buffer = bigNum.sub(bnS).toArrayLike(Buffer);
+        signature.s = new Uint8Array(buffer);
+    }
+
+    return signature;
+}
+
+export function _checkMalleability(signature: { r: Uint8Array; s: Uint8Array }, curveParams: { name: string }) {
+    let halfOrder = undefined;
+    switch (curveParams.name) {
+        case "P-256":
+            halfOrder = ordersForCurve.secp256r1.halfOrder;
+            break;
+        case "P-384":
+            halfOrder = ordersForCurve.secp384r1.halfOrder;
+            break;
+        default:
+            break;
+    }
```

Comment:
This seems to be duplicated code as this exact piece is also used in the `preventMalleability` method. Consider using an external method for it.